File: virksomhetssertifikat-core/src/main/java/no/ks/fiks/virksomhetsertifikat/Sertifikat.java
package no.ks.fiks.virksomhetsertifikat;

import lombok.Data;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;

@Data
public class Sertifikat {
    @NotNull
    private SertifikatType sertifikatType;
    @NotNull
    private String keystorePassword;
    @NotBlank
    private String keystorePath;
    @NotNull
    private String certificateAlias;
    @NotBlank
    private String privateKeyAlias;
    @NotNull
    private String privateKeyPassword;
}

File: virksomhetssertifikat-core/src/main/kotlin/no/ks/fiks/virksomhetsertifikat/Sertifikat.kt
package no.ks.fiks.virksomhetsertifikat

data class Sertifikat(
    val sertifikatType: SertifikatType,
    val keystorePassword: String,
    val keystorePath: String,
    val certificateAlias: String,
    val privateKeyAlias: String,
    val privateKeyPassword: String
)